config/jobs/kubernetes-sigs/testing_frameworks/testing-frameworks-config.yaml

```diff
@@ -5,15 +5,32 @@ presubmits:
     - master
     always_run: true
     skip_report: false
+    decorate: true
     labels:
       preset-service-account: "true"
+      preset-dind-enabled: "true"
     spec:
```

Comment:
> I am not sure what is needed to actually run docker on prow:

The way to think about this is "what is needed to run docker on kubernetes?" since prow just schedules a pod and doesn't otherwise provide a runtime environment.

Note that you are writing a podspec here. Anything you can put into a podspec, you can put into here

By default prow doesn't do anything aside from picking a name for your pod and kubectl create -f this-pod.yaml

Reply:
I understand that it is "just a pod". What I was wondering was which automatism prow provides, what does e.g. `preset-dind-enabled: "true"` do, what is and how does this decoration work, and which image do I need to use.

Thanks for all your comments, helped already a lot answering all those questions!

Reply:
preset-dind enabled enables docker in docker when used in conjunction with the correct image, see here: https://github.com/kubernetes/test-infra/blob/b4a0e4f20212565ccbfabdbd0ddee6aa40b05aea/prow/config.yaml#L538

it's a pod-preset, but with prow handling it first class for :reasons:

You generally don't need this enabled for jobs, as you can just schedule an image that includes the tools you need as your podspec. See Erick's comments below for what decorate does.